src/app/shared/axiosHelper.js

```diff
@@ -1,4 +1,60 @@
+const axios = require("axios");
+const https = require("https");
+const { getCriFromErrorResponse } = require("./loggerHelper");
+
+const createAxiosInstance = (baseUrl) => {
+  const instance = axios.create({
+    baseURL: baseUrl,
+    httpsAgent: new https.Agent({ keepAlive: true }),
+  });
+
+  instance.interceptors.response.use(null, axiosErrorHandler);
+
+  return instance;
+};
+
+const axiosErrorHandler = (error) => {
+  const logger = error.config.logger;
+
+  if (axios.isAxiosError(error)) {
+    if (error.response) {
+      const cri = getCriFromErrorResponse(error.response);
+
+      const message = {
+        description: "Error response received from API",
+        errorMessage: error.message,
+        endpoint: `${error.request?.method} ${error.request?.path}`,
+        data: error.response.data,
+      };
+
+      if (cri) {
+        message.cri = cri;
```

Comment:
If you want a 'clever' one-liner, you can do:
```
description: "Error response received from API",
errorMessage: error.message,
endpoint: `${error.request?.method} ${error.request?.path}`,
data: error.response.data,
...(cri && { cri }),
```
...but I think the explicit `if` is a lot clearer!

Reply:
I prefer the explicit if, more readable 😃